lib/apiservers/engine/backends/container.go

```diff
@@ -793,7 +793,81 @@ func (c *Container) ContainerUpdate(name string, hostConfig *containertypes.Host
 // timeout, an error is returned. If you want to wait forever, supply
 // a negative duration for the timeout.
 func (c *Container) ContainerWait(name string, timeout time.Duration) (int, error) {
-	return 0, fmt.Errorf("%s does not implement container.ContainerWait", ProductName())
+	defer trace.End(trace.Begin(fmt.Sprintf("name(%s):timeout(%s)", name, timeout)))
+
+	// Look up the container name in the metadata cache to get long ID
+	vc := cache.ContainerCache().GetContainer(name)
+	if vc == nil {
+		return -1, NotFoundError(name)
+	}
+	id := vc.ContainerID
+
+	//retrieve client to portlayer
+	client := c.containerProxy.Client()
+	results, err := client.Containers.ContainerWait(containers.NewContainerWaitParamsWithContext(ctx).WithTimeout(int64(timeout.Seconds())).WithID(id))
+	if err != nil {
+		switch err := err.(type) {
+		case *containers.ContainerWaitNotFound:
```

Comment:
I'm assuming you're referring to the persona container cache -- if so, event activity is not streamed to the persona yet, so that cache wouldn't be mutated based on events.

So -- in the case that the container isn't found on the PL side then yes...we should eject it from the persona side.  I can add that when I rebase.